DynamicTablesPkg/Library/Common/Tpm2DeviceTableLib/Tpm2DeviceTableLib.c

```diff
@@ -0,0 +1,329 @@
+/** @file
+  Tpm2 device table generating Library
+
+  Copyright (c) 2025, Arm Limited. All rights reserved.<BR>
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  @par Reference(s):
+  - TCG ACPI specification.
+    (https://developer.arm.com/documentation/ihi0069/)
+**/
+#include <IndustryStandard/DebugPort2Table.h>
+#include <Library/AcpiLib.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/PcdLib.h>
+
+#include <Protocol/AcpiTable.h>
+
+// Module specific include files.
+#include <AcpiTableGenerator.h>
+#include <ConfigurationManagerObject.h>
+#include <ConfigurationManagerHelper.h>
+#include <Library/AcpiHelperLib.h>
+#include <Library/AmlLib/AmlLib.h>
+#include <Protocol/ConfigurationManagerProtocol.h>
+
+/** C array containing the compiled AML template.
+    This symbol is defined in the auto generated C file
+    containing the AML bytecode array.
+*/
+extern CHAR8  tpm2devicetabletemplate_aml_code[];
+
+/** Fixup the Tpm2 device UID (_UID).
+
+  @param  [in]  RootNodeHandle  Pointer to the root of an AML tree.
+  @param  [in]  Uid             UID for the Serial Port.

```

Comment:
Should search for `serial` word in the file, I think some comments should be updated